resolver/addressable_resolver.go

```diff
@@ -163,7 +169,8 @@ func (r *URIResolver) URIFromObjectReference(ctx context.Context, ref *corev1.Ob
 		return url, nil
 	}
 
-	_, lister, err := r.informerFactory.Get(ctx, gvr)
+	lister, err := r.listerFactory(gvr)
+
```

Comment:
super-nit: I'd drop this newline (to keep the error handling together with the function that returns the err), and add back the one after the `ref == nil` guard clause (so after L149) that got dropped here

```suggestion
```